src/main/java/uk/ac/ebi/embl/converter/fftogff3/GFF3AnnotationFactory.java

```diff
@@ -343,20 +340,28 @@ private String getParentFeature(String emblFeatureName, Optional geneName) {
         return "";
     }
 
-    private String getGFF3FeatureName(Feature ffFeature) {
+    private String getGFF3FeatureName(Feature ffFeature) throws ValidationException {
 
-        List<ConversionEntry> mappings = ConversionUtils.getFF2GFF3FeatureMap().get(ffFeature.getName());
+        String featureName = ffFeature.getName();
+        List<ConversionEntry> mappings = ConversionUtils.getFF2GFF3FeatureMap().get(featureName);
         if (mappings == null) {
-            return ffFeature.getName();
+            RuleSeverityState.handleValidationException(new UnmappedFFFeatureException(featureName));
+            return featureName;
```

Comment:
We are throwing UnmappedFFFeatureException 2 times in this implementation. Can't we unify it to throw once when feature + qualifier don't match?